### PR TITLE
fix: disable rename feature when strictTemplates is disabled

### PR DESCRIPTION
**Pull request by atscott**
When `strictTemplates` is disabled, we know that the compiler's view of a template
is not completely "accurate". That is, the compiler's generated Type
Check Blocks will not have sufficient type information to properly
perform a rename operation.

### CODE REVIEW

File: server/src/session.ts [hunk 1]
```diff
@@ -49,6 +49,12 @@ export class Session {
   private readonly logToConsole: boolean;
   private diagnosticsTimeout: NodeJS.Timeout|null = null;
   private isProjectLoading = false;
+  /**
+   * By default, the rename feature is enabled. However, if we detect the compiler options
+   * diagnostic that suggests enabling strict mode, we want to disable renaming because we know that
+   * there are many cases where it will not work correctly.
+   */
+  private disableRenaming = false;
```
Comment (kyliau) [inline server/src/session.ts:57]: This should be a per-project setting, so we probably want a `WeakMap<ts.server.Project, boolean>` here.

Comment (atscott) [reply server/src/session.ts:57]: Good catch - I made this a `Set<ts.server.Project>` to just track which projects have renaming disabled.

File: server/src/session.ts [hunk 2]
```diff
@@ -49,6 +49,13 @@ export class Session {
   private readonly logToConsole: boolean;
   private diagnosticsTimeout: NodeJS.Timeout|null = null;
   private isProjectLoading = false;
+  /**
+   * Tracks which `ts.server.Project`s have the renaming capability disabled.
+   *
+   * If we detect the compiler options diagnostic that suggests enabling strict mode, we want to
+   * disable renaming because we know that there are many cases where it will not work correctly.
+   */
+  private renameDisabledProjects: Set<ts.server.Project> = new Set();
```
Comment (kyliau) [inline server/src/session.ts:58]: Should we use a `WeakSet` here so that projects that are closed can be gc-ed?

Comment (atscott) [reply server/src/session.ts]: 👍  Done.

**Review by kyliau — APPROVED** [review]

**Pull request merged**